File: step2_conferencia.py
"""
step2_conferencia.py
Carrega boletos_pdf.json e compara com o banco Novacorp (idimovel=970).
Busca no banco SOMENTE pelos IDs presentes no PDF + a inadimplencia agregada do BI.
"""
import sys, io, re, json
sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')

import mysql.connector
from datetime import datetime

JSON_PATH  = r"boletos_pdf.json"
DB_CFG     = dict(host="sistemasnovacorp.com.br", port=5643,
                  database="novacorpconect", user="Intelligence",
                  password="@bv2026@")
ID_EMPRESA = 75
ID_IMOVEL  = 970

def fmt_brl(v):
    v = float(v or 0)
    s = f"{v:,.2f}".replace(",","X").replace(".",",").replace("X",".")
    return f"R$ {s}"

def rc(v): return round(float(v or 0), 2)
SEP = "=" * 82

# ── Carrega PDF ───────────────────────────────────────────────────────────────
with open(JSON_PATH, encoding="utf-8") as f:
    todos_pdf = json.load(f)

normais_pdf  = [b for b in todos_pdf if not b["is_credito"]]
creditos_pdf = [b for b in todos_pdf if b["is_credito"]]
ids_pdf      = [b["idboleto"] for b in todos_pdf]
ids_norm_pdf = set(b["idboleto"] for b in normais_pdf)

print(f"\n{SEP}")
print("  CONFERENCIA: PDF Coqueiros x Banco Novacorp (idimovel=970)")
print(f"  Data: {datetime.now().strftime('%d/%m/%Y %H:%M')}")
print(f"{SEP}\n")
print(f"PDF: {len(normais_pdf)} boletos normais + {len(creditos_pdf)} creditos")

# ── Banco ─────────────────────────────────────────────────────────────────────
print("Conectando ao banco...")
conn = mysql.connector.connect(**DB_CFG)
cur  = conn.cursor(dictionary=True)

# 1. Busca dados dos IDs do PDF (em lotes de 500 para evitar pacotes grandes)
print(f"Buscando {len(ids_pdf)} boletos no banco por ID...")
banco_por_id = {}
LOTE = 500
for i in range(0, len(ids_pdf), LOTE):
    lote = ids_pdf[i:i+LOTE]
    placeholders = ",".join(["%s"] * len(lote))
    cur.execute(f"""
        SELECT idboleto, idcliente, pago, cancelado, origem,
               total, valorparc, dataVecto, dataPgto,
               DATE_FORMAT(dataVecto,'%%Y-%%m') AS mesRef
        FROM TBBOLETO
        WHERE idEmpresa=%s AND idboleto IN ({placeholders})
    """, [ID_EMPRESA] + lote)
    for r in cur.fetchall():
        banco_por_id[r["idboleto"]] = r
    print(f"  Lote {i//LOTE+1}: {len(banco_por_id)} encontrados ate agora...")

# 2. Inadimplencia agregada do BI (query exata do route.ts)
print("Buscando sumario de inadimplencia do BI...")
cur.execute("""
    SELECT DATE_FORMAT(b.dataVecto,'%%Y-%%m') AS mesRef,
           SUM(IFNULL(b.valorparc,0)) AS valor,
           COUNT(*) AS qtd
    FROM TBBOLETO b
    WHERE b.idEmpresa=%s AND b.idimovel=%s
      AND b.pago=0 AND b.cancelado=0
      AND b.dataVecto < CURDATE()
    GROUP BY mesRef HAVING valor>0
    ORDER BY mesRef DESC
""", (ID_EMPRESA, ID_IMOVEL))
bi_sumario = {r["mesRef"]: r for r in cur.fetchall()}

# 3. IDs na inadimplencia do BI para este imovel
cur.execute("""
    SELECT idboleto, valorparc, dataVecto, DATE_FORMAT(dataVecto,'%%Y-%%m') AS mesRef
    FROM TBBOLETO
    WHERE idEmpresa=%s AND idimovel=%s
      AND pago=0 AND cancelado=0
      AND dataVecto < CURDATE()
""", (ID_EMPRESA, ID_IMOVEL))
bi_inad_rows = cur.fetchall()
bi_inad_ids  = {r["idboleto"]: r for r in bi_inad_rows}

cur.close()
conn.close()
print(f"Banco: {len(banco_por_id)} boletos encontrados pelos IDs do PDF")
print(f"Banco: {len(bi_inad_ids)} boletos na inadimplencia do BI\n")

# ── Totais por mes ────────────────────────────────────────────────────────────
pdf_por_mes = {}
for b in normais_pdf:
    m = b["mes_ref"]
    pdf_por_mes.setdefault(m, {"valor":0.0, "qtd":0})
    pdf_por_mes[m]["valor"] = rc(pdf_por_mes[m]["valor"] + b["valor_pdf"])
    pdf_por_mes[m]["qtd"]  += 1

bi_por_mes = {}
for r in bi_inad_rows:
    m = r["mesRef"]
    bi_por_mes.setdefault(m, {"valor":0.0, "qtd":0})
    bi_por_mes[m]["valor"] = rc(bi_por_mes[m]["valor"] + rc(r["valorparc"]))
    bi_por_mes[m]["qtd"]  += 1

# ── Tabela comparativa ────────────────────────────────────────────────────────
print(f"{SEP}")
print("  COMPARACAO POR MES  —  PDF vs BI (inadimplencia, query route.ts)")
print(f"{SEP}")
print(f"  {'Mes':>7} | {'PDF':>14} {'Qtd':>4} | {'BI Banco':>14} {'Qtd':>4} | {'Dif':>14} | Status")
print(f"  {'─'*80}")

todos_meses = sorted(set(list(pdf_por_mes.keys()) + list(bi_por_mes.keys())), reverse=True)
tp = tb = 0.0
div_meses = []

for mes in todos_meses:
    vp = pdf_por_mes.get(mes, {"valor":0.0})["valor"]
    qp = pdf_por_mes.get(mes, {"qtd":0})["qtd"]
    vb = bi_por_mes.get(mes, {"valor":0.0})["valor"]
    qb = bi_por_mes.get(mes, {"qtd":0})["qtd"]
    d  = rc(vp - vb)
    tp += vp; tb += vb
    st = "OK" if abs(d)<0.02 else "*** DIVERGE ***"
    if abs(d)>=0.02: div_meses.append(mes)
    print(f"  {mes:>7} | {fmt_brl(vp):>14} {qp:>4} | {fmt_brl(vb):>14} {qb:>4} | {fmt_brl(d):>14} | {st}")

print(f"  {'─'*80}")
td = rc(tp - tb)
print(f"  {'TOTAL':>7} | {fmt_brl(tp):>14}      | {fmt_brl(tb):>14}      | {fmt_brl(td):>14} |")

# ── No PDF mas NAO no BI ──────────────────────────────────────────────────────
hoje = datetime.now()
no_pdf_nao_bi = [b for b in normais_pdf if b["idboleto"] not in bi_inad_ids]

print(f"\n{SEP}")
print(f"  BOLETOS NO PDF mas NAO na inadimplencia do BI  ({len(no_pdf_nao_bi)} itens)")
print(f"{SEP}")
if no_pdf_nao_bi:
    print(f"  {'idboleto':>9} | {'mes':>7} | {'vecto':>12} | {'val_pdf':>13} | Motivo no banco")
    print(f"  {'─'*85}")
    for b in no_pdf_nao_bi:
        idb = b["idboleto"]
        if idb in banco_por_id:
            r = banco_por_id[idb]
            motivos = []
            if r["pago"]      == 1:
                dp = str(r["dataPgto"])[:10] if r["dataPgto"] else "?"
                motivos.append(f"PAGO em {dp}")
            if r["cancelado"] == 1: motivos.append("CANCELADO")
            if r["dataVecto"] and r["dataVecto"] >= hoje: motivos.append("Nao vencido")
            if not motivos: motivos.append(f"pago={r['pago']} canc={r['cancelado']}")
            st = " + ".join(motivos)
            print(f"  {idb:>9} | {b['mes_ref']:>7} | {b['vecto']:>12} | {fmt_brl(b['valor_pdf']):>13} | {st}")
            print(f"           banco: valorparc={fmt_brl(r['valorparc'])} total={fmt_brl(r['total'])} orig={r['origem']} cli={r['idcliente']}")
        else:
            print(f"  {idb:>9} | {b['mes_ref']:>7} | {b['vecto']:>12} | {fmt_brl(b['valor_pdf']):>13} | NAO ENCONTRADO NO BANCO")
else:
    print("  Nenhum.")

# ── No BI mas NAO no PDF ──────────────────────────────────────────────────────
no_bi_nao_pdf = [idb for idb in bi_inad_ids if idb not in ids_norm_pdf]

print(f"\n{SEP}")
print(f"  BOLETOS na inadimplencia do BI mas NAO no PDF  ({len(no_bi_nao_pdf)} itens)")
print(f"{SEP}")
if no_bi_nao_pdf:
    print(f"  {'idboleto':>9} | {'mesRef':>7} | {'dataVecto':>12} | {'valorparc':>13} | idcliente")
    print(f"  {'─'*75}")
    for idb in sorted(no_bi_nao_pdf):
        r = bi_inad_ids[idb]
        print(f"  {idb:>9} | {r['mesRef']:>7} | {str(r['dataVecto'])[:10]:>12} | {fmt_brl(r['valorparc']):>13} |")
else:
    print("  Nenhum.")

# ── Valores diferentes (em ambos mas valor diverge) ──────────────────────────
div_valor = []
for b in normais_pdf:
    idb = b["idboleto"]
    if idb in bi_inad_ids:
        r   = bi_inad_ids[idb]
        vb  = rc(r["valorparc"])
        vp  = rc(b["valor_pdf"])
        if abs(vb - vp) >= 0.02:
            div_valor.append((b, r))

print(f"\n{SEP}")
print(f"  BOLETOS EM AMBOS com VALOR DIFERENTE  ({len(div_valor)} itens)")
print(f"  PDF.valor_pdf vs banco.valorparc")
print(f"{SEP}")
if div_valor:
    print(f"  {'idboleto':>9} | {'mes':>7} | {'val_pdf':>13} | {'valorparc':>13} | {'dif':>13} | unidade")
    print(f"  {'─'*85}")
    for b, r in div_valor:
        d = rc(b["valor_pdf"] - rc(r["valorparc"]))
        print(f"  {b['idboleto']:>9} | {b['mes_ref']:>7} | {fmt_brl(b['valor_pdf']):>13} | {fmt_brl(r['valorparc']):>13} | {fmt_brl(d):>13} | {b['unidade']}")
else:
    print("  Nenhum — todos os boletos em comum tem valores iguais.")

# ── Creditos/estornos ─────────────────────────────────────────────────────────
if creditos_pdf:
    print(f"\n{SEP}")
    print(f"  CREDITOS/ESTORNOS NO PDF  ({len(creditos_pdf)} itens)")
    print(f"{SEP}")
    print(f"  {'idboleto':>9} | {'mes':>7} | {'vecto':>12} | {'val_pdf':>13} | unidade")
    print(f"  {'─'*75}")
    total_cred = 0.0
    for b in creditos_pdf:
        print(f"  {b['idboleto']:>9} | {b['mes_ref']:>7} | {b['vecto']:>12} | {fmt_brl(b['valor_pdf']):>13} | {b['unidade']}")
        if b["idboleto"] in banco_por_id:
            r = banco_por_id[b["idboleto"]]
            print(f"           banco: valorparc={fmt_brl(r['valorparc'])} pago={r['pago']} canc={r['cancelado']} orig={r['origem']}")
        total_cred += b["valor_pdf"]
    print(f"  Total creditos: {fmt_brl(total_cred)}")

print(f"\n{SEP}\n")

File: step2_conferencia.js
// step2_conferencia.js
// Carrega boletos_pdf.json e compara com banco Novacorp (idimovel=970)
// Usa mysql2 que ja funciona bem no projeto.

process.env.MYSQL_HOST     = "sistemasnovacorp.com.br";
process.env.MYSQL_PORT     = "5643";
process.env.MYSQL_DATABASE = "novacorpconect";
process.env.MYSQL_USER     = "Intelligence";
process.env.MYSQL_PASSWORD = "@bv2026@";

const fs    = require("fs");
const mysql = require("mysql2/promise");

const ID_EMPRESA = 75;
const ID_IMOVEL  = 970;
const JSON_PATH  = "./boletos_pdf.json";

function rc(v)    { return Math.round((parseFloat(v) || 0) * 100) / 100; }
function fmtBRL(v){ return "R$ " + rc(v).toFixed(2).replace(".", ",").replace(/\B(?=(\d{3})+(?!\d))/g, "."); }
const SEP = "=".repeat(80);

async function main() {
  // ── Carrega JSON do PDF ──────────────────────────────────────────────────
  const todosPDF    = JSON.parse(fs.readFileSync(JSON_PATH, "utf-8"));
  const normaisPDF  = todosPDF.filter(b => !b.is_credito);
  const creditosPDF = todosPDF.filter(b => b.is_credito);
  const idsPDF      = todosPDF.map(b => b.idboleto);
  const idsNormSet  = new Set(normaisPDF.map(b => b.idboleto));

  console.log(`\n${SEP}`);
  console.log("  CONFERENCIA: PDF Coqueiros x Banco Novacorp (idimovel=970)");
  console.log(`  ${new Date().toLocaleString("pt-BR")}`);
  console.log(`${SEP}\n`);
  console.log(`PDF: ${normaisPDF.length} boletos normais + ${creditosPDF.length} creditos`);

  // ── Conecta ao banco ─────────────────────────────────────────────────────
  console.log("Conectando ao banco...");
  const conn = await mysql.createConnection({
    host: process.env.MYSQL_HOST,
    port: Number(process.env.MYSQL_PORT),
    database: process.env.MYSQL_DATABASE,
    user: process.env.MYSQL_USER,
    password: process.env.MYSQL_PASSWORD,
    timezone: "+00:00",
    connectTimeout: 30000,
  });
  console.log("Conectado!\n");

  // ── Query 1: IDs do PDF no banco ─────────────────────────────────────────
  // Divide em lotes para o IN() não ser enorme
  const bancoPorId = {};
  const LOTE = 300;
  for (let i = 0; i < idsPDF.length; i += LOTE) {
    const lote = idsPDF.slice(i, i + LOTE);
    const [rows] = await conn.query(
      `SELECT idboleto, idcliente, pago, cancelado, origem,
              total, valorparc, dataVecto, dataPgto,
              DATE_FORMAT(dataVecto,'%Y-%m') AS mesRef
       FROM TBBOLETO
       WHERE idEmpresa=? AND idboleto IN (${lote.map(()=>"?").join(",")})`,
      [ID_EMPRESA, ...lote]
    );
    rows.forEach(r => { bancoPorId[r.idboleto] = r; });
    process.stdout.write(`  Lote ${Math.floor(i/LOTE)+1}: ${Object.keys(bancoPorId).length} encontrados\r`);
  }
  console.log(`\nBanco (pelos IDs do PDF): ${Object.keys(bancoPorId).length} boletos encontrados`);

  // ── Query 2: Inadimplencia do BI (query exata route.ts) ──────────────────
  console.log("Buscando inadimplencia do BI...");
  const [inadRows] = await conn.query(
    `SELECT idboleto, valorparc, dataVecto, DATE_FORMAT(dataVecto,'%Y-%m') AS mesRef
     FROM TBBOLETO
     WHERE idEmpresa=? AND idimovel=?
       AND pago=0 AND cancelado=0
       AND dataVecto < CURDATE()`,
    [ID_EMPRESA, ID_IMOVEL]
  );
  const biInadById = {};
  inadRows.forEach(r => { biInadById[r.idboleto] = r; });
  console.log(`BI inadimplencia: ${inadRows.length} boletos\n`);

  await conn.end();

  // ── Totais por mes ────────────────────────────────────────────────────────
  const pdfPorMes = {};
  normaisPDF.forEach(b => {
    if (!pdfPorMes[b.mes_ref]) pdfPorMes[b.mes_ref] = { valor: 0, qtd: 0 };
    pdfPorMes[b.mes_ref].valor = rc(pdfPorMes[b.mes_ref].valor + b.valor_pdf);
    pdfPorMes[b.mes_ref].qtd++;
  });

  const biPorMes = {};
  inadRows.forEach(r => {
    if (!biPorMes[r.mesRef]) biPorMes[r.mesRef] = { valor: 0, qtd: 0 };
    biPorMes[r.mesRef].valor = rc(biPorMes[r.mesRef].valor + rc(r.valorparc));
    biPorMes[r.mesRef].qtd++;
  });

  // ── Tabela comparativa ────────────────────────────────────────────────────
  console.log(`${SEP}`);
  console.log("  COMPARACAO POR MES  —  PDF vs BI");
  console.log(`${SEP}`);
  console.log(`  ${"Mes".padStart(7)} | ${"PDF Valor".padStart(15)} ${"Qtd".padStart(4)} | ${"BI Valor".padStart(15)} ${"Qtd".padStart(4)} | ${"Dif".padStart(15)} | Status`);
  console.log(`  ${"─".repeat(80)}`);

  const todosMeses = [...new Set([...Object.keys(pdfPorMes), ...Object.keys(biPorMes)])].sort().reverse();
  let tp = 0, tb = 0;
  const divMeses = [];

  for (const mes of todosMeses) {
    const vp = pdfPorMes[mes]?.valor || 0;
    const qp = pdfPorMes[mes]?.qtd   || 0;
    const vb = biPorMes[mes]?.valor  || 0;
    const qb = biPorMes[mes]?.qtd    || 0;
    const d  = rc(vp - vb);
    tp += vp; tb += vb;
    const st = Math.abs(d) < 0.02 ? "OK" : "*** DIVERGE ***";
    if (Math.abs(d) >= 0.02) divMeses.push(mes);
    console.log(`  ${mes.padStart(7)} | ${fmtBRL(vp).padStart(15)} ${String(qp).padStart(4)} | ${fmtBRL(vb).padStart(15)} ${String(qb).padStart(4)} | ${fmtBRL(d).padStart(15)} | ${st}`);
  }
  console.log(`  ${"─".repeat(80)}`);
  const td = rc(tp - tb);
  console.log(`  ${"TOTAL".padStart(7)} | ${fmtBRL(tp).padStart(15)}      | ${fmtBRL(tb).padStart(15)}      | ${fmtBRL(td).padStart(15)} |`);

  // ── No PDF mas NAO no BI ──────────────────────────────────────────────────
  const noPDFnaoBI = normaisPDF.filter(b => !biInadById[b.idboleto]);
  console.log(`\n${SEP}`);
  console.log(`  BOLETOS NO PDF mas NAO na inadimplencia do BI  (${noPDFnaoBI.length} itens)`);
  console.log(`${SEP}`);
  if (noPDFnaoBI.length) {
    // Agrupa por motivo para facilitar leitura
    const porMotivo = { PAGO: [], CANCELADO: [], NAO_ENCONTRADO: [], OUTRO: [] };
    for (const b of noPDFnaoBI) {
      const r = bancoPorId[b.idboleto];
      if (!r) { porMotivo.NAO_ENCONTRADO.push(b); continue; }
      if (r.pago == 1) { porMotivo.PAGO.push({b, r}); continue; }
      if (r.cancelado == 1) { porMotivo.CANCELADO.push({b, r}); continue; }
      porMotivo.OUTRO.push({b, r});
    }

    console.log(`\n  PAGOS (${porMotivo.PAGO.length}): estao no PDF como inadimplentes mas foram pagos no banco`);
    if (porMotivo.PAGO.length) {
      console.log(`  ${"idboleto".padStart(9)} | ${"mes".padStart(7)} | ${"dataPgto".padStart(12)} | ${"val_pdf".padStart(13)} | unidade`);
      console.log(`  ${"─".repeat(80)}`);
      for (const {b,r} of porMotivo.PAGO) {
        const dp = r.dataPgto ? String(r.dataPgto).substring(0,10) : "?";
        console.log(`  ${String(b.idboleto).padStart(9)} | ${b.mes_ref.padStart(7)} | ${dp.padStart(12)} | ${fmtBRL(b.valor_pdf).padStart(13)} | ${b.unidade}`);
      }
    }

    console.log(`\n  CANCELADOS (${porMotivo.CANCELADO.length}): estao no PDF mas estao cancelados no banco`);
    if (porMotivo.CANCELADO.length) {
      console.log(`  ${"idboleto".padStart(9)} | ${"mes".padStart(7)} | ${"val_pdf".padStart(13)} | unidade`);
      for (const {b,r} of porMotivo.CANCELADO) {
        console.log(`  ${String(b.idboleto).padStart(9)} | ${b.mes_ref.padStart(7)} | ${fmtBRL(b.valor_pdf).padStart(13)} | ${b.unidade}`);
      }
    }

    console.log(`\n  NAO ENCONTRADOS NO BANCO (${porMotivo.NAO_ENCONTRADO.length})`);
    if (porMotivo.NAO_ENCONTRADO.length) {
      for (const b of porMotivo.NAO_ENCONTRADO) {
        console.log(`  ${String(b.idboleto).padStart(9)} | ${b.mes_ref.padStart(7)} | ${fmtBRL(b.valor_pdf).padStart(13)} | ${b.unidade}`);
      }
    }

    console.log(`\n  OUTRO/INDEFINIDO (${porMotivo.OUTRO.length})`);
    if (porMotivo.OUTRO.length) {
      for (const {b,r} of porMotivo.OUTRO) {
        console.log(`  ${String(b.idboleto).padStart(9)} | ${b.mes_ref.padStart(7)} | pago=${r.pago} canc=${r.cancelado} | ${b.unidade}`);
      }
    }
  } else {
    console.log("  Nenhum.");
  }

  // ── No BI mas NAO no PDF ──────────────────────────────────────────────────
  const noBInaoDF = inadRows.filter(r => !idsNormSet.has(r.idboleto));
  console.log(`\n${SEP}`);
  console.log(`  BOLETOS na inadimplencia do BI mas NAO no PDF  (${noBInaoDF.length} itens)`);
  console.log(`${SEP}`);
  if (noBInaoDF.length) {
    console.log(`  ${"idboleto".padStart(9)} | ${"mesRef".padStart(7)} | ${"dataVecto".padStart(12)} | ${"valorparc".padStart(13)}`);
    console.log(`  ${"─".repeat(60)}`);
    for (const r of noBInaoDF.sort((a,b)=>a.idboleto-b.idboleto)) {
      console.log(`  ${String(r.idboleto).padStart(9)} | ${r.mesRef.padStart(7)} | ${String(r.dataVecto).substring(0,10).padStart(12)} | ${fmtBRL(r.valorparc).padStart(13)}`);
    }
  } else {
    console.log("  Nenhum.");
  }

  // ── Valores diferentes ────────────────────────────────────────────────────
  const divValor = normaisPDF.filter(b => {
    const r = biInadById[b.idboleto];
    return r && Math.abs(rc(b.valor_pdf) - rc(r.valorparc)) >= 0.02;
  });

  console.log(`\n${SEP}`);
  console.log(`  BOLETOS EM AMBOS com VALOR DIFERENTE  (${divValor.length} itens)`);
  console.log(`${SEP}`);
  if (divValor.length) {
    console.log(`  ${"idboleto".padStart(9)} | ${"mes".padStart(7)} | ${"val_pdf".padStart(13)} | ${"valorparc".padStart(13)} | ${"dif".padStart(13)} | unidade`);
    console.log(`  ${"─".repeat(85)}`);
    for (const b of divValor) {
      const r  = biInadById[b.idboleto];
      const vb = rc(r.valorparc);
      const d  = rc(b.valor_pdf - vb);
      console.log(`  ${String(b.idboleto).padStart(9)} | ${b.mes_ref.padStart(7)} | ${fmtBRL(b.valor_pdf).padStart(13)} | ${fmtBRL(vb).padStart(13)} | ${fmtBRL(d).padStart(13)} | ${b.unidade}`);
    }
  } else {
    console.log("  Nenhum — valores iguais em todos boletos em comum.");
  }

  // ── Creditos ──────────────────────────────────────────────────────────────
  if (creditosPDF.length) {
    console.log(`\n${SEP}`);
    console.log(`  CREDITOS/ESTORNOS NO PDF  (${creditosPDF.length} itens)`);
    console.log(`${SEP}`);
    let totalCred = 0;
    console.log(`  ${"idboleto".padStart(9)} | ${"mes".padStart(7)} | ${"vecto".padStart(12)} | ${"val_pdf".padStart(13)} | banco (pago/canc/orig) | unidade`);
    console.log(`  ${"─".repeat(90)}`);
    for (const b of creditosPDF) {
      const r = bancoPorId[b.idboleto];
      const banco_info = r ? `pago=${r.pago} canc=${r.cancelado} orig=${r.origem}` : "NAO ENCONTRADO";
      console.log(`  ${String(b.idboleto).padStart(9)} | ${b.mes_ref.padStart(7)} | ${b.vecto.padStart(12)} | ${fmtBRL(b.valor_pdf).padStart(13)} | ${banco_info} | ${b.unidade}`);
      totalCred += b.valor_pdf;
    }
    console.log(`\n  Total creditos PDF: ${fmtBRL(totalCred)}`);
  }

  console.log(`\n${SEP}\n`);
}

main().catch(err => {
  console.error("\nErro:", err.message);
  process.exit(1);
});
